src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -4426,11 +4428,14 @@ private StatementAst UsingStatementRule(Token usingToken)
             if (itemAst == null)
             {
                 ReportError(itemToken.Extent, () => ParserStrings.InvalidValueForUsingItemName, itemToken.Text);
+                // ErrorRecovery: If there is no identifier, skip whole 'using' line
+                SyncOnError(true, TokenKind.NewLine);
```

Comment:
`SyncOnError` skips tokens until it finds the one(s) you say are where to stop. Semicolon terminates a using statement in the same way a newline does, so we should assume we saw the end of a using statement if we see a semicolon or newline.